x/metadata/types/specification_test.go

```diff
@@ -0,0 +1,497 @@
+			true,
+		},
+		// Simple valid case
+		{
+			"simple valid case",
+			NewScopeSpecification(
+				ScopeSpecMetadataAddress(uuid.New()),
+				nil,
+				[]string{specTestBech32},
+				[]PartyType{PartyType_PARTY_TYPE_OWNER},
+				[]MetadataAddress{GroupSpecMetadataAddress(uuid.New())},
+			),
+			"",
+			false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		s.T().Run(tt.name, func(t *testing.T) {
+			err := tt.spec.ValidateBasic()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Scope ValidateBasic error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.wantErr {
+				require.Equal(t, tt.want, err.Error())
+			}
+		})
+	}
+}
+
+func (s *specificationTestSuite) TestDescriptionValidateBasic() {
+	tests := []struct {
+		name     string
+		desc     *Description
+		want     string
+		wantErr  bool
+	}{
+		// Name tests
+		{
+			"invalid name - empty",
+			NewDescription(
+				"",
+				"",
+				"",
+				"",
+			),
+			fmt.Sprintf("description Name cannot be empty"),
+			true,
+		},
+		{
+			"invalid name - too long",
+			NewDescription(
+				strings.Repeat("x", maxDescriptionNameLength + 1),
+				"",
+				"",
+				"",
+			),
+			fmt.Sprintf("description Name exceeds maximum length (expected <= %d got: %d)", maxDescriptionNameLength, maxDescriptionNameLength + 1),
+			true,
+		},
+		{
+			"valid name - 1 char",
+			NewDescription(
+				"x",
+				"",
+				"",
+				"",
+			),
+			"",
+			false,
+		},
+		{
+			"valid name - exactly max length",
+			NewDescription(
+				strings.Repeat("y", maxDescriptionNameLength),
+				"",
+				"",
+				"",
+			),
+			"",
+			false,
+		},
+
+		// Description tests
+		{
+			"invalid description - too long",
+			NewDescription(
+				"Unit Tests",
+				strings.Repeat("z", maxDescriptionDescriptionLength + 1),
+				"",
+				"",
+			),
+			fmt.Sprintf("description Description exceeds maximum length (expected <= %d got: %d)", maxDescriptionDescriptionLength, maxDescriptionDescriptionLength + 1),
+			true,
+		},
+		{
+			"valid description - empty",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"",
+				"",
+			),
+			"",
+			false,
+		},
+		{
+			"valid description - 1 char",
+			NewDescription(
+				"Unit Tests",
+				"z",
+				"",
+				"",
+			),
+			"",
+			false,
+		},
+		{
+			"valid description - exactly max length",
+			NewDescription(
+				"Unit Tests",
+				strings.Repeat("z", maxDescriptionDescriptionLength),
+				"",
+				"",
+			),
+			"",
+			false,
+		},
+
+		// Website url tests
+		{
+			"invalid website url - too long",
+			NewDescription(
+				"Unit Tests",
+				"",
+				strings.Repeat("h", maxURLLength + 1),
+				"",
+			),
+			fmt.Sprintf("url WebsiteUrl exceeds maximum length (expected <= %d got: %d)", maxURLLength, maxURLLength + 1),
+			true,
+		},
+		{
+			"invalid website url - no protocol",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"www.test.com",
+				"",
+			),
+			fmt.Sprintf("url WebsiteUrl must begin with either http:// or https://"),
```

Comment:
see related comment above re: schemes.  Data URLs (presuming they fit in the length restriction) would be quite useful here.

```
data:image/png;base64,iVBORw0KGgoAAAANSU[...]AAAASUVORK5CYII=
```